### DISCUSSION TITLE
MQTT Publish

### DISCUSSION THREAD

**Discussion by bdkacz** [basking-in-the-sun2000/solar-logger#31] [General]
Hi.

Is it possible to add function to publish via MQTT current power and daily production ? 

Home Assistant add new Energy management   function  https://www.home-assistant.io/blog/2021/08/04/home-energy-management/  

,but it doesn't work with influxdb sensor as  production value 

Maybe use http://www.steves-internet-guide.com/publishing-messages-mqtt-client/

**Comment by basking-in-the-sun2000** [comment]
It is possible to use mqtt. However since the data is already in influxdb, you could have another process send the data to the broker, or add extra data with Telegraf.

Could you let me know what you are trying to do? Seems you solved this, but curious about your application.

**Comment by bdkacz** [comment]
I wanted to do this because after a few hours of trying HA did not want to correctly interpret the influxdb database data. But since I finally succeeded, I closed the issue

**Comment by WMP** [comment]
Could you told me how you solve this?

**Comment by bdkacz** [comment]
> Could you told me how you solve this?

I create influxdb sensor based on https://www.home-assistant.io/integrations/influxdb/ 

with query

  queries:
    - name: Huawei_Production
      group_function: last
      where: '1=1'
      measurement: '"Huawei"'
      field: P_accum
      database: logger  

and  template sensor using this influxdb sensor   with attributes   
        unit_of_measurement: "kWh"
        device_class: energy
        state_class: total_increasing

then I can add created template sensor to energy panel as value of production

**Comment by f432** [comment]
Has anyone been able to setup exported energy into Home Assistant using influxdb as a sensor? I've spent days on it but my knowledge is not there,

Thought I would try here before asking in the HA forums.

**Comment by f432** [comment]
so to answer my own question, i just used the existing influxdb sensor "total grid exported" and just added the following to show up in the energy template of HA:

device_class: energy
state_class: total_increasing
